tls/extensions/s2n_client_key_share.c

```diff
@@ -102,25 +103,17 @@ static int s2n_generate_default_ecc_key_share(struct s2n_connection *conn, struc
     return S2N_SUCCESS;
 }
 
-static int s2n_generate_default_pq_hybrid_key_share(struct s2n_connection *conn, struct s2n_stuffer *out) {
+static int s2n_generate_pq_hybrid_key_share(struct s2n_connection *conn, struct s2n_stuffer *out,
+        struct s2n_kem_group_params *kem_group_params) {
     notnull_check(conn);
     notnull_check(out);
+    notnull_check(kem_group_params);
 
-    if (s2n_is_in_fips_mode()) {
-        /* PQ KEMs are not supported in FIPS mode */
-        return S2N_SUCCESS;
-    }
+    /* This function should never be called when in FIPS mode */
+    ENSURE_POSIX(s2n_is_in_fips_mode() == false, S2N_ERR_PQ_KEMS_DISALLOWED_IN_FIPS);
```

Comment:
I'm worried that these "not fips" checks are scattered across all the PQ code. What if you later decide to make PQ available for fips? Wouldn't this be cleaner / easier to manage if you just checked for fips mode once when enabling PQ, or maybe wrote a getter that included the check for any fields that shouldn't be set in fips mode?

Reply:
Agreed. I think we need cleaner, more generalized checks for PQ support that encompass the FIPS checks and `S2N_NO_PQ` flags. I've created https://github.com/awslabs/s2n/issues/2285 and will follow up on that when I get some time.